File: thih/erlang/infer.erl
% // 11 Type Inference
% object Infer {
%   import Pred._
%   import Assump._
%   import TIMonad._
% 
%   type Infer[E, T] = Ti => ClassEnv => List[Assump] => E => (List[Pred], T)
% }
% 

File: thih/ocaml2/infer.ml
(* 11 Type Inference *)
(*|
# Infer

    >>> open Infer;;
    
*)

open Pred
open Assump
open TIMonad

type ('e, 't) infer = ti -> classEnv -> assump list -> 'e -> pred list * 't

(*|

*)
